### PR TITLE
Added Deepcover command [client-2016]

### PR DESCRIPTION
**Pull request by coderuk**
Allows admins to change names without affecting original name on relogin.

### CODE REVIEW

**Comment by coderuk** [comment]
not sure why label is failing?

**Comment by QuentinGruber** [comment]
> not sure why label is failing?

It's not because of you don't worry :)

File: src/servers/ZoneServer2016/commands/commands.ts [hunk 1]
```diff
@@ -2155,7 +2155,62 @@ export const commands: Array<Command> = [
...
+      }
+
+      // Update the client's character name
+      client.character.name = newCharacterName;
 
+      // Wait for one second before running vanish command
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // Set the client's isSpectator state
+      client.character.isSpectator = !client.character.isSpectator;
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/commands/commands.ts:2185]: But if it's already set to true it will be falsy then ? Is it intentionnal ?

**Review by QuentinGruber — CHANGES_REQUESTED** [review]

File: src/servers/ZoneServer2016/commands/commands.ts (hunk 1 above)
Comment (coderuk) [reply src/servers/ZoneServer2016/commands/commands.ts:2187]: yes is intentional to force the name change to become active

File: src/servers/ZoneServer2016/commands/commands.ts [hunk 2]
```diff
@@ -2155,7 +2155,62 @@ export const commands: Array<Command> = [
...
+              iteratedClient.spawnedEntities.indexOf(client.character),
+              1
+            );
+          }
+        }
+        server.sendData(client, "Spectator.Enable", {});
+      }
+
+      // Wait for an additional second before running the second vanish command
+      await new Promise((resolve) => setTimeout(resolve, 1000));
```
[the remark this answers is not in the document]
Comment (coderuk) [reply src/servers/ZoneServer2016/commands/commands.ts]: these changes are now completed

**Review by QuentinGruber — APPROVED** [review]

**Pull request merged**